libsolidity/formal/BMC.cpp

```diff
@@ -1227,8 +1214,9 @@ BMC::checkSatisfiableAndGenerateModel(std::vector<smtutil::Expression> const& _e
 	{
 		if (m_settings.printQuery)
 		{
-			auto portfolio = dynamic_cast<smtutil::SMTPortfolio*>(m_interface.get());
-			std::string smtlibCode = portfolio->dumpQuery(_expressionsToEvaluate);
+			auto smtlibInterface = dynamic_cast<smtutil::SMTLib2Interface*>(m_interface.get());
```

Comment:
Maybe it would be good to get rid off `SolverInterface` class and use `SMTLib2Interface`? 

That makes sense if we plan to use only SMTLib2 compatible solvers.